runtime/compilers/rillv1/parser.go

```diff
@@ -185,8 +185,9 @@ type Parser struct {
 	Errors    []*runtimev1.ParseError
 
 	// Internal state
-	resourcesForPath           map[string][]*Resource // Reverse index of Resource.Paths
-	resourcesForUnspecifiedRef map[string][]*Resource // Reverse index of Resource.rawRefs where kind=ResourceKindUnspecified
+	resourcesForPath           map[string][]*Resource    // Reverse index of Resource.Paths
+	resourcesForUnspecifiedRef map[string][]*Resource    // Reverse index of Resource.rawRefs where kind=ResourceKindUnspecified
+	resourceNamesForDataPaths  map[string][]ResourceName // Index of local data files to resources that depend on them
```

Comment:
How will this get garbage collected? The other index maps here are cleared in `deleteResource` ([link](https://github.com/rilldata/rill/blob/24626af89fb7642801a7d612d4a7c9cf5ee8c0cd/runtime/compilers/rillv1/parser.go#L860))

Reply:
Though I think we may still need to clean the `resourceNamesForDataPaths`. I think iterating the full map should be okay. It is not expected to be a very large map (only populated for local_files).

Reply:
Yeah I didn't mean memory garbage collection, just making sure the values are dropped from the map when they stop being correct :)